lldb/source/Plugins/TypeSystem/Swift/SwiftASTContext.cpp

```diff
@@ -1752,17 +1745,16 @@ SwiftASTContext::CreateInstance(lldb::LanguageType language, Module &module,
       swift_ast_sp->m_module_import_warnings.push_back(std::string(error));
     }
 
-    // Some of the bits in the compiler options we keep separately, so
-    // we need to populate them from the serialized options:
     llvm::StringRef serialized_triple =
         swift_ast_sp->GetCompilerInvocation().getTargetTriple();
-    if (serialized_triple.empty()) {
-      LOG_PRINTF(LIBLLDB_LOG_TYPES, "Serialized triple was empty.");
-    } else {
-      LOG_PRINTF(LIBLLDB_LOG_TYPES, "Found serialized triple %s.",
+    if (!serialized_triple.empty()) {
+      LOG_PRINTF(LIBLLDB_LOG_TYPES, "Serialized/default triple would have been %s.",
                  serialized_triple.str().c_str());
-      swift_ast_sp->SetTriple(llvm::Triple(serialized_triple), &module);
-      set_triple = true;
+      // The serialized triple is the triple of the last binary
+      // __swiftast section that was processed. Instead of relying on
+      // the section contents order, we overwrite the triple in the
+      // CompilerInvocation with the triple recovered from the binary.
+      swift_ast_sp->SetTriple(llvm::Triple(triple), &module);
```

Comment:
is this supposed to be `llvm::Triple(serialized_triple)`?

Reply:
No, this is overwriting the deserialized triple with the one loaded from the binary. See comment.